ftw/referencewidget/resources/refwidget.js

```diff
@@ -0,0 +1,167 @@
+            item['selectable'] = true;
+            item['selected'] = 'checked="checked"';
+            item['type'] = sel_type;
+            item['checkbox'] = checkbox_template(item);
+            $(container).find('ul').append(list_template(item));
+        });
+    });
+
+    function openOverlay(e){
+        $('.refbrowser').remove();
+        list_template = Handlebars.compile($('#listing-template').html());
+        checkbox_template = Handlebars.compile($('#checkbox-template').html());
+        $('.sortable').sortable();
+        target = $(e.currentTarget);
+        sel_type = target.closest('.referencewidget').data('type')
+        field_id = target.closest('.field').attr('id');
+        name = target.closest('.field').data('fieldname');
+        widget_url = target.closest('.referencewidget').data('url') + "/++widget++" + name;
+        $('body').append($("#refbrowser-template").html());
+        build_pathbar("");
+        get_data("");
+//        build_list(request_data);
+    }
+
+function search(e){
+    var value = $(e.currentTarget.parentNode).find('input:text').val()
+    $.post(widget_url + '/search_for_refs', {'term': value}, function(data){
+        rebuild_listing(JSON.parse(data));
+    })
+}
+
+function build_pathbar(path){
+    $('.refbrowser .path').empty();
+    var pathbar = $.post(widget_url +'/generate_pathbar', {'origin': path}, function(data){
+    var hb_template = Handlebars.compile($("#node-template").html());
+    var pathbar = '';
+    JSON.parse(data).forEach(function(item){
+        pathbar += hb_template(item);
+    });
+    $('.path').append(pathbar);
+
+    });
+
+}
+
+function switch_level(e){
+    var ident = $(e.currentTarget).data('id');
+    var path = request_data['items'][lookup_table[ident]]['path']
+    request_path = path;
+    build_pathbar(path);
+    get_data(path);
+    }
+
+function change_page(e){
+    $('.refbrowser_batching .batch').removeClass("active_batch");
+    $(e.currentTarget).addClass("active_batch");
+    get_data(request_path, parseInt($(e.currentTarget).text()));
+}
+
+function rebuild_listing(data){
+    $('.refbrowser .listing ul').empty();
+    var list_html = "";
+    lookup_table = {}
+    for (var key in data) {
+        if (data.hasOwnProperty(key)) {
+            var item = data[key];
+            lookup_table[item['id']] = key;
+            item['selected'] = "";
+            var is_selected = $('.referencewidget .selected_items li[data-path="' + item['path'] + '"]');
+            if (is_selected.length > 0) {
+                item['selected'] = 'checked="checked"';
+            }
+
+            if (item['selectable']){
+                item['type'] = sel_type;
+                item['checkbox'] = checkbox_template(item);
+            }
+            list_html += list_template(item);
+        }
+    }
+    $('.listing ul').append(list_html);
+
+}
+
+function build_list(data){
+            $('.refbrowser .refbrowser_batching').remove();
```

Comment:
The indenting is completely inconsistent. Please configure your editor so that it helps you to indent properly 😉